test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Formatter/ODataEntityReferenceLinkE2ETests.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Data.Entity;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.OData.Edm;
+using Microsoft.Test.E2E.AspNet.OData.Common.Controllers;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.Formatter
+{
+    public class ODataEntityReferenceLinkE2ETests : WebHostTestBase
+    {
+        public ODataEntityReferenceLinkE2ETests(WebHostTestFixture fixture)
+            : base(fixture)
+        {
+        }
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            var controllers = new[] { typeof(BooksController)};
+            configuration.AddControllers(controllers);
+            configuration.Count().Filter().OrderBy().Expand().MaxTop(null).Select();
+            configuration.MapODataServiceRoute("odata", "odata", BuildEdmModel(configuration));
+        }
+        private static IEdmModel BuildEdmModel(WebRouteConfiguration configuration)
+        {
+            var builder = configuration.CreateConventionModelBuilder();
+            builder.EntitySet<Book>("Books");
+            builder.EntitySet<Author>("Authors");
+            builder.Action("ResetDataSource");
+            builder.Action("RelateToExistingEntityAndUpdate").ReturnsFromEntitySet<Book>("Books").EntityParameter<Book>("book");
+
+            return builder.GetEdmModel();
+        }
+
+        [Fact]
+        public async Task CanCreate_ANewEntityAndRelateToAnExistingEntity_UsingODataBind()
+        {
+            await ResetDataSource();
+            // Arrange
+            const string Payload = "{" +
+            "\"Id\":\"1\"," +
+            "\"Name\":\"BookA\"," +
+            "\"Author@odata.bind\":\"Authors(1)\"}";
+
+            string Uri = BaseAddress + "/odata/Books";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, Uri);
+
+            request.Content = new StringContent(Payload);
+            request.Content.Headers.ContentType = MediaTypeWithQualityHeaderValue.Parse("application/json");
+
+            // Act
+            HttpResponseMessage response = await Client.SendAsync(request);
```

Comment:
This also implements `IDisposable`. Please check for other types in the tests as well